lib/player.js

```diff
@@ -3798,7 +3802,21 @@ shaka.Player = class extends shaka.util.FakeEventTarget {
   addTextTrack(uri, language, kind, mime, codec, label) {
     // TODO: Add an actual error for this.
     if (this.loadMode_ == shaka.Player.LoadMode.SRC_EQUALS) {
-      shaka.log.error('Cannot add text when loaded with src=');
+      const trackElement = document.createElement('track');
+      trackElement.setAttribute('src', uri);
+      trackElement.setAttribute('label', label || '');
+      trackElement.setAttribute('kind', kind);
+      trackElement.setAttribute('srclang', language);
+      this.video_.appendChild(trackElement);
+      const textTracks = this.getTextTracks();
+      const track = textTracks.find((t) => t.language == language);
```

Comment:
I wonder if language is the best thing to check for.  Certainly multiple tracks could be added with the same language and either different MIME types or roles.  What if instead you check for `t.src == uri`?

Reply:
I added more checks. The uri is not a valid check because `shaka.extern.Track` doesn't include it

Reply:
Ah, okay.  That makes sense.  I was thinking of the HTML5 Track object.